fastjet/internal/delaunay/triangle.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+
+	"go-hep.org/x/hep/fastjet/internal/predicates"
+)
+
+// Triangle is a set of three points that make up a triangle. It stores hierarchical information to find triangles.
+type Triangle struct {
+	// children are triangles that lead to the removal of this triangle.
+	// When a point is inserted the triangle that contains the point is found by going down the hierarchical tree.
+	// The tree's root is the root triangle and the children slice contains the children triangles.
+	children []*Triangle
+	// A,B,C are the CCW-oriented points that make up the triangle
+	A, B, C *Point
```

Comment:
If these points were in `[3]*Point`, I think a lot of the switch casing could be reduced to a loop. This would simplify the code at various places. For example in findNearest.